docs/developers/setup.md

```diff
@@ -0,0 +1,132 @@
+# Development environment
+
+<!-- markdownlint-disable MD046 -->
+
+This section outlines some suggestions as well as conventions used by the EMMOntoPy developers, which should be considered or followed if one wants to contribute to the package.
+
+## Setup
+
+!!! important "Requirements"
+    This section expects you to be running on a Unix-like system (e.g., Linux) with *minimum* Python 3.6.
+
+### Virtual environment
+
+Since development can be messy, it is good to separate the development environment from the rest of your system's environment.
+
+To do this, you can use a virtual environment.
+There are a several different ways to create a virtual environment, but we recommend using either `conda` or `virtualenv`.
+
+=== "`virtualenv` (recommended)"
+
+    To install `virtualenv` run:
+
+    ```console
+    pip install virtualenvwrapper
+    ```
+
+    There is other setup, most of which only needs to be run once.
+    For more information about this, see the [`virtualenvwrapper` documentation](https://virtualenvwrapper.readthedocs.io/en/latest/#introduction).
+
+    After successfully setting up `virtualenv` through `virtualenvwrapper`, you can create a new virtual environment:
+
+    ```console
+    mkproject -p python3.7 emmo-python
+    ```
+
+    !!! note
+        If you do not have Python 3.7 installed (but instead have Python 3.6), you can leave out the extra `-p python3.7` argument.
+        Or you can choose to use another version of Python by changing this argument to another (valid) python interpreter.
+
+    Then, if the virtual environment has not been activated automatically (you should see the name `emmo-python` in a parenthesis in your console), you can run:
+
+    ```console
+    workon emmo-python
+    ```
+
+    To deactivate the virtual environment, returning to the system/global environment again, run:
+
+    ```console
+    deactivate
+    ```
+
```

Comment:
```suggestion
=== "`venv`"
    First go to the directory where you desire to keep you virtual environment and choosing
    the python interpreter you already have installed and wish to use run
    ```console
    python3.? -m venv emmo-python
    ```
    A subfolder with the name emmo-python containing the environment is created.
    To activate the environment
    ```console
    path/to/python-environments/emmo-python/activate
    ```
    Deactivate in the same manner as for virtualenv
```